### PR TITLE
Add structured logger

### PR DESCRIPTION
**Pull request by bpeng**
## Proposed Changes

Changes proposed in this pull request:

- Add structured logger to be shared in other projects

## Production Changes

The following production changes are required to deploy these changes:

- None

## Review

Check the box that applies to this code review.  If necessary please seek help with adding a checklist guide for the reviewer.
When assigning the code review please consider the expertise needed to review the changes.

- [ ] This is a content (documentation, web page etc) only change.
- [x] This is a minor change (meta data, bug fix, improve test coverage etc).
- [ ] This is a larger change (new feature, significant refactoring etc).  Please use the code review guidelines to add a checklist below to guide the code reviewer.

### Code Review Guide

*Insert check list here if needed.*
This file is currently under [haz](https://github.com/GeoNet/haz/blob/main/internal/log/logger.go)

Usage example:

[logger.Info("Received heartbeat ", "ServiceID", h.ServiceID)](https://github.com/GeoNet/haz/blob/main/internal/msg/heartbeat.go#L29)

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package log
```
Comment (CallumNZ) [inline log/logger.go:1]: I don't think we should name this the same as the standard library's `log` package.

https://go.dev/blog/package-names
`Avoid unnecessary package name collisions. While packages in different directories may have the same name, packages that are frequently used together should have distinct names. This reduces confusion and the need for local renaming in client code. For the same reason, avoid using the same name as popular standard packages like io or http.`

But then I don't know what it should be called 😅

File: log/logger.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		opts = &slog.HandlerOptions{Level: slog.LevelError}
+	case "WARN":
+		opts = &slog.HandlerOptions{Level: slog.LevelWarn}
+	case "DEBUG":
+		opts = &slog.HandlerOptions{Level: slog.LevelDebug}
+	default:
+		opts = &slog.HandlerOptions{Level: slog.LevelInfo}
+	}
+	logger = slog.New(slog.NewTextHandler(os.Stdout, opts))
+}
```
Comment (CallumNZ) [inline log/logger.go:28]: Can we replace all `init` magic to a function?

Something like `NewLogger`

File: log/logger.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// args: expected in key:value pairs
+func Warn(msg string, args ...any) {
+	logger.Warn(msg, args...)
+}
+
+// encapsulating func in log/slog
+// args: expected in key:value pairs
+func Error(msg string, args ...any) {
+	logger.Error(msg, args...)
+}
```
Comment (CallumNZ) [inline log/logger.go:52]: Should these be methods on the logger struct?

e.g.:

```go
func (logger *slog.Logger) Error(msg string, args ...any) {
    logger.Error(msg, args...)
}
```

**Review by CallumNZ — COMMENTED** [review]
I've left some comments, but a bigger picture question is, do we need this? Can we not just use `slog` directly?

File: log/logger.go (hunk 3 above)
Comment (junghao) [reply log/logger.go:52]: @CallumNZ I think the usage of `log` or `logger` is not via methods, we'll just use it as an exported function from the package. 

Take an example,  we invoke `log.Println()` in place,  without `mylogger := log.NewLogger()` then call `mylogger.Error()`.

Comment (CallumNZ) [reply log/logger.go]: `Take an example, we invoke log.Println() in place, without mylogger := log.NewLogger() then call mylogger.Error().`

I don't understand.

Comment (junghao) [reply log/logger.go]: NVM I read it wrong. LOL

Comment (junghao) [reply log/logger.go]: Example of this approach:

https://github.com/GeoNet/field-monitor-project/blob/main/internal/fmp_logger/fmp_logger.go

Then we simply call the package's method without initializing
https://github.com/GeoNet/field-monitor-project/blob/main/cmd/fmp-polling/main.go#L84

**Pull request closed without merging**